fe/fe-core/src/main/java/com/starrocks/transaction/OlapTableTxnStateListener.java

```diff
@@ -82,6 +82,12 @@ public void preCommit(TransactionState txnState, List<TabletCommitInfo> tabletCo
         Map<Long, Set<Long>> tabletToBackends = new HashMap<>();
         Set<Long> allCommittedBackends = new HashSet<>();
 
+        // 1. record tablet commit infos in TransactionState,
+        // so we can decide to update version in replica when finish transaction
+        if (!tabletCommitInfos.isEmpty()) {
+            txnState.setTabletCommitInfos(tabletCommitInfos);
+        }
```

Comment:
No need to set the tablet commit info list in `LakeTableTxnStateListener`?

Reply:
Not need, there is not clone in lake table.

Reply:
But how do you make sure the tablet commit info is not needed in shared data mode

Reply:
tablet commit info was also present in shared data mode before, try to be consistent unless you're sure it's something that should be removed in shared data mode, and if so, then you should also commit a separate patch to remove it

Reply:
Yes, I add tablet commit info  in shared data mode before, but in fact it's useless in shared data mode.

Reply:
Pull requests should do only one thing. Following the single responsibility principle reduces the overhead caused by revising a code that attempts to solve several problems.

Reply:
It's hard for reviewers to tell if it's okay to remove tablet commit info from the transaction state in shared data mode.